src/environment/environment.py

```diff
@@ -0,0 +1,165 @@
+
+        Args:
+            screen (Surface): Pygame screen surface.
+            agent_size (int, optional): Size of the agent. Defaults to 10.
+            render_on (bool, optional): Whether to render the environment. Defaults to False.
+        """
+        self.screen = screen
+        self.render_on = render_on
+        self.agent_size = agent_size
+
+        self.agent_location = (
+            (self.screen.get_size()[0] / 2),
+            (self.screen.get_size()[1] / 2),
+        )
+
+        self.rewards = {
+            "alive": 1,
+            "death": -100,
+            "not possible": -5,
+        }
+
+        self.actions = {
+            "up": 0,
+            "down": 1,
+            "left": 2,
+            "right": 3,
+        }
+
+    def reset(self) -> np.ndarray:
+        """
+        Resets the environment.
+
+        Returns:
+            np.ndarray: Initial state of the environment.
+        """
+        self.agent_location = (
+            (self.screen.get_size()[0] / 2),
+            (self.screen.get_size()[1] / 2),
+        )
+
+        # TODO: reset enemies; self.enemy_generator.reset()? Dunno if needed
+
+        if self.render_on:
+            self.render()
+
+        return self.get_state()
+
+    def render(self) -> None:
+        """
+        Renders the environment.
+        """
+        self.screen.fill((0, 0, 0))
+        pygame.draw.circle(
+            self.screen, (255, 255, 255), self.agent_location, self.agent_size
+        )
+        # TODO: draw enemeis; self.enemies.render()?
+        pygame.display.flip()
+
+    def get_state(self) -> np.ndarray:
+        """
+        Gets the current state of the environment.
+
+        Returns:
+            np.ndarray: Current state of the environment.
+        """
+        # TODO: Add the closest enemy state here
+        # Something like enemy_location = self.enemies.get_closest_enemy(self.agent_location, ...)
+        state = np.array([self.agent_location[0], self.agent_location[1]])
+
+        # TODO: if enemy state is added:
+        # state = np.array([self.agent_location[0], self.agent_location[1], enemy_location[0], enemy_location[1]])
+        return state
+
+    def move_agent(self, action: int) -> Tuple[int, bool]:
+        """
+        Moves the agent based on the action taken.
+
+        Args:
+            action (int): Action to be taken.
+
+        Returns:
+            Tuple[int, bool]: Reward obtained and whether the episode is done.
+        """
+        moves = {
+            self.actions["up"]: (0, -1),
+            self.actions["down"]: (0, 1),
+            self.actions["left"]: (-1, 0),
+            self.actions["right"]: (1, 0),
+        }
+
+        previous_location = self.agent_location
+
+        move = moves.get(action)
+        new_location = (previous_location[0] + move[0], previous_location[1] + move[1])
+
+        done = False
+        reward = self.rewards["alive"]
+
+        if self.is_valid_location(new_location):
```

Comment:
Shouldn't this be 
```suggestion
        if not self.is_valid_location(new_location):
```
?